python/pyarrow/tests/parquet/test_datetime.py

```diff
@@ -297,6 +297,8 @@ def test_coerce_int96_timestamp_unit(unit):
 
 @pytest.mark.pandas
 @pytest.mark.parametrize('pq_reader_method', ['ParquetFile', 'read_table'])
+@pytest.mark.filterwarnings(
+    "ignore:Discarding nonzero nanoseconds in conversion:UserWarning")
```

Comment:
Where is this warning coming from?

Reply:
Haivng a hard time getting it to show up agaiin locally, but it's iin CI: https://github.com/milesgranger/arrow/actions/runs/3573348441/jobs/6007303891#step:6:5679

I _think_ it's coming from `tab_correct.to_pandas` call. But not sure, will try more to get it to happen locally.

Reply:
Searching for this phrase, pandas raises this warning when you convert a Timestamp object to a python datetime.datetime object:

```
In [22]: pd.Timestamp("2022-01-01 01:01:01.123456789").to_pydatetime()
<ipython-input-22-7cef94b12f00>:1: UserWarning: Discarding nonzero nanoseconds in conversion.
  pd.Timestamp("2022-01-01 01:01:01.123456789").to_pydatetime()
Out[22]: datetime.datetime(2022, 1, 1, 1, 1, 1, 123456)
```

Reply:
But I don't directly see how this test is causing such a warning, since it uses dates that doesn't include nanoseconds

Reply:
Ah, or maybe from this line:

```
    # with the default resolution of ns, we get wrong values for INT96
    # that are out of bounds for nanosecond range
    tab_error = get_table(pq_reader_method, filename)
    assert tab_error["a"].to_pylist() != oob_dts
```

since you get "wrong" values, those have overflown, and the resulting wrong datetime will probably have nanoseconds, and thus can raise this warning. That should be OK to ignore then.

Reply:
Should I move to a `warnings.catch` over this line specifically then to make it more clear where the warning is being raised?

Reply:
> Should I move to a `warnings.catch` over this line specifically then to make it more clear where the warning is being raised?

That's probably most specific indeed (and also documents the root cause as well)

Reply:
https://github.com/apache/arrow/pull/14729/commits/ab4a31a4a1dbca63f8e35eb3553f8b43f9d0f71a